src/pages/find-task-page.tsx

```diff
@@ -75,18 +76,32 @@ const FindTaskPage: FC<FindTaskPageProps & FindTaskActions> = (props) => {
         latitudeDelta: 5,
         longitudeDelta: 5,
     }
-    
+
     // Bottom sheet
     const bottomSheetRef = useRef<BottomSheet>(null);
     const bottomBarPos = useRef(new Animated.Value(0));
     const [bottomSheetIsOpen, setBottomSheetIsOpen] = useState(true);
-    
+
     // Layout
     const insets = useSafeArea();
     const windowHeight = Dimensions.get('window').height;
     const bottomSheetMaxHeight = windowHeight * 0.5;
     const bottomSheetMinHeight = 50 + insets.bottom;
 
+    useFocusEffect(
+        React.useCallback(() => {
+            const task = InteractionManager.runAfterInteractions(() => {
+                if (userLocation) {
+                    props.searchForNearbyTasks({
+                        coordinate: { latitude: userLocation.latitude, longitude: userLocation.longitude },
+                        radius: searchDistance
+                    });
+                }
+            });
+            return () => task.cancel();
+        }, [])
+    );
```

Comment:
```suggestion
    useFocusEffect(
        React.useCallback(() => {
            const task = InteractionManager.runAfterInteractions(() => {
                if (userLocation) {
                    props.searchForNearbyTasks({
                        coordinate: { latitude: userLocation.latitude, longitude: userLocation.longitude },
                        radius: searchDistance
                    });
                }
            });
            return () => task.cancel();
        }, [(userLocation === null)])
    );
```
When I go to Find Task and click on a task and then accept it, it is still left in the list when I go back to Find Task. Does this not happen on iOS?

The problem seems to be the if-statement. If the userLocation hasn't been set yet, it will just continue without ever re-searching the task. But if I put a check for userLocation === null in the dependency list, it works for me! Could you check if it works for you?